File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        unordered_map<char, char> bracket = {
        {')', '('},
        {']', '['},
        {'}', '{'}
    };
        stack<char> arr;
        for(auto c: s){
            if(c=='('||c=='{'||c=='[') arr.push(c);
            else{
                if (arr.empty() || arr.top() != bracket[c]) return false;
                arr.pop();
            }
        }
        return arr.empty();
    }
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        arr = []
        pairs = {
            ')' : '(',
            '}' : '{',
            ']' : '['
        }
        for i in s:
            if i in '({[': 
                arr.append(i)
            elif arr and pairs[i] == arr[-1]:
                arr.pop()
            else:
                return False
        return not arr